pkg/chaosdaemon/dns_server.go

```diff
@@ -45,10 +45,12 @@ func (s *DaemonServer) SetDNSServer(ctx context.Context,
 		}
 
 		// backup the /etc/resolv.conf
-		cmd := bpm.DefaultProcessBuilder("sh", "-c", fmt.Sprintf("ls %s.chaos.bak || cp %s %s.chaos.bak", DNSServerConfFile, DNSServerConfFile, DNSServerConfFile)).
-			SetNS(pid, bpm.MountNS).
-			SetContext(ctx).
-			Build()
+		processBuilder := bpm.DefaultProcessBuilder("sh", "-c", fmt.Sprintf("ls %s.chaos.bak || cp %s %s.chaos.bak", DNSServerConfFile, DNSServerConfFile, DNSServerConfFile)).SetContext(ctx)
+		if !req.WithoutNS {
```

Comment:
Why not add a `EnterNS` filed instead of `WithoutNS`?

Reply:
the default value is false. if use EnterNS, I must modify every request in chaos mesh

Reply:
I prefer `EnterNS`(or further `EnterNSForContainer`), the default value is evil. 
I think it's necessary to declare explicitly that we need to enter the target namespaces of a specific container.

Reply:
OK, I update `WithoutNS` to `EnterNS`